src/libcoretest/num/mod.rs

```diff
@@ -177,4 +177,61 @@ mod tests {
     test_impl_from! { test_u16i32, u16, i32 }
     test_impl_from! { test_u16i64, u16, i64 }
     test_impl_from! { test_u32i64, u32, i64 }
+
+    // Signed -> Float
+    test_impl_from! { test_i8f32, i8, f32 }
```

Comment:
Thanks for elaborating.  I don't have a specific example already where this would be useful, but it's not hard to concoct one.  Say `log<F: Into<f64>>(self, base: F) -> f64`, where you want the flexibility of float but the base is often integral.  Yes, I think this could accept `u8`, but that's not really implicit since the function used `Into`, opting into that flexibility.  The "type safety prevents mistakes" argument doesn't speak to me much, because you could say the same about even integral conversions, and again `Into` is an opt-in choice.

But I realize this is a pretty subjective thing to judge.  Is there anyone else we should ping for an opinion about int->float?  Do you at least agree with f32->f64?